templates/httpboot_dhcp_class.erb

```diff
@@ -0,0 +1,7 @@
+match if substring(option vendor-class-identifier, 0, 10) = "HTTPClient";
+  if option architecture = 00:0F {
+    filename "<%= @proxy_base_url %>/EFI/grub2/shimia32.efi";
+  } else if option architecture = 00:10 {
+    filename "<%= @proxy_base_url %>/EFI/grub2/shim.efi";
+  }
+  option vendor-class-identifier "HTTPClient"
```

Comment:
I need to see this rendered, however the "match" statement must be used within a "class" declaration which I don't see here. It's a weird syntax indeed. Alternatively, I can rewrite the config to more traditional "if then" pattern but the weird looking match syntax actually looks nicer than nested ifs.

Reply:
I think rewriting the existing rules for PXEClients into class-match and providing new class-match block for HTTPClients is the cleanest result.

Reply:
https://github.com/theforeman/puppet-dhcp/blob/master/manifests/dhcp_class.pp isn't the greatest, but it uses https://github.com/theforeman/puppet-dhcp/blob/master/templates/dhcpd.class.erb. It always adds a semi-colon and 2 spaces indent which is why the first line has the odd indenting and I moved the option to the last line. with that, the DHCP server at least started up for me.

Reply:
Cool, how can I test this on a production setup?